javascript/workflow/src/resources/DataPackage.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-assignment */
+import type { DataPackageSchema } from '@datashaper/schema'
+import { createDataPackageSchemaObject } from '@datashaper/schema'
+import type { TableContainer } from '@datashaper/tables'
+import { introspect } from '@datashaper/tables'
+import type { Maybe } from '@datashaper/workflow'
+import { Workflow } from '@datashaper/workflow'
+import type { Observable, Subscription } from 'rxjs'
+import { BehaviorSubject } from 'rxjs'
+import { map } from 'rxjs/operators'
+
+import { Codebook } from './Codebook.js'
+import type { DataSource } from './DataSource.js'
+import { Named } from './Named.js'
+import type { SchemaResource } from './types.js'
+
+export class DataPackage
+	extends Named
+	implements SchemaResource<DataPackageSchema>
+{
+	private readonly _output = new BehaviorSubject<Maybe<TableContainer>>(
+		undefined,
+	)
+
+	private _codebook?: any
+	private _outputSubscription?: Subscription
+
+	public constructor(
+		id: string,
+		public readonly source: DataSource,
+		public readonly workflow = new Workflow(),
```

Comment:
Do you think this is sufficiently generic enough to handle any Resource type? It almost seems like we should have a list() and get(name) type of model?

Reply:
E.g., versus hard-coded workflow and codebook instances...

Reply:
Well, the problem here is that in app-code we need to get the live instances, not just the generic schema variant. The data-package is kind of a weird outlier when it comes to serialize/deserialize from JSON because it's so abstract resource oriented.

Reply:
Maybe we can address this later with a plugin model. Ideally you'd have a factory function that you can register profiles with, which return live objects. So the data package won't really care about hard-coded instances - it will have a list of Resources, and retrieving one invokes the right factory, then passes it to the right component that knows how to render it...